WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailNewHeaderView.swift

```diff
@@ -0,0 +1,421 @@
+import SwiftUI
+import WordPressUI
+import Gridicons
+
+/// A protocol conformed by both `ReaderDetailHeaderView` and `ReaderDetailNewHeaderViewHost`.
+/// This is a temporary solution to abstract and simplify usage in the view controller.
+///
+/// NOTE: This protocol should be removed once the `readerImprovements` flag is deleted.
+///
+protocol ReaderDetailHeader: NSObjectProtocol {
+    var delegate: ReaderDetailHeaderViewDelegate? { get set }
+    var useCompatibilityMode: Bool { get set }
+
+    func configure(for post: ReaderPost)
+    func refreshFollowButton()
+}
+
+// MARK: - SwiftUI View Host
+
+class ReaderDetailNewHeaderViewHost: UIView {
+    weak var delegate: ReaderDetailHeaderViewDelegate?
+
+    // TODO: Find out if we still need this.
+    var useCompatibilityMode: Bool = false
+
+    private let isLoggedIn: Bool
+
+    // TODO: Populate this with values from the ReaderPost.
+    private lazy var viewModel: ReaderDetailHeaderViewModel = {
+        $0.topicDelegate = self
+        $0.headerDelegate = delegate
+        return $0
+    }(ReaderDetailHeaderViewModel())
```

Comment:
On my end, this gets initialized before the `delegate` for `ReaderDetailNewHeaderViewHost` is set. Which causes all the delegate functions in the view model to fail because the `headerDelegate` is `nil`.

Reply:
Thanks, Chris. Yeah, I think I messed up my commits. I stumbled upon this exact problem, and I've already fixed it... but it didn't get included in this PR. This is now fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21674/commits/1bce13ec0b5c746294c6b50e534acb674953508c.